File: frontend/src/app/shared/models/meeting/meeting.ts
import { Participant } from '../participant/participant';
import { PollResultDto } from '../poll/poll-result-dto';
import { Recurrence } from './meeting-recurrence';

export interface Meeting {
  id: string;
  topic?: string;
  description?: string;
  settings: string;
  startTime: Date;
  endTime?: Date;
  anonymousCount: number;
  isScheduled: boolean;
  isRecurrent: boolean;
  isWhiteboard: boolean;
  isPoll: boolean;
  isAllowedToChooseRoom: boolean;
  recognitionLanguage: string;
  recurrence: Recurrence;
  isAudioAllowed: boolean;
  isVideoAllowed: boolean;
  speechDuration: number;
  presenceDuration: number;
  participants: Participant[];
  pollResults: PollResultDto[];
}

File: backend/Whale.DAL/Models/Meeting.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Whale.DAL.Abstraction;
using Polls = Whale.DAL.Models.Poll;

namespace Whale.DAL.Models
{
    public class Meeting : BaseEntity
    {
        public string Settings { get; set; }
        public string Topic { get; set; }
        public string Description { get; set; }
        public DateTimeOffset StartTime { get; set; }
        public DateTimeOffset? EndTime { get; set; }
        public int AnonymousCount { get; set; }
        public bool IsScheduled { get; set; }
        public bool IsRecurrent { get; set; }

        [NotMapped]
        public IEnumerable<Participant> Participants { get; set; }

        public IEnumerable<Polls.Poll> PollResults { get; set; }
        public IEnumerable<AgendaPoint> AgendaPoints { get; set; }
        public Meeting() { }

        public Meeting(Meeting meeting)
        {
            Id = meeting.Id;
            Settings = meeting.Settings;
            StartTime = meeting.StartTime;
            EndTime = meeting.EndTime;
            AnonymousCount = meeting.AnonymousCount;
            IsScheduled = meeting.IsScheduled;
            IsRecurrent = meeting.IsRecurrent;

            Participants = meeting.Participants;
            PollResults = meeting.PollResults;
        }

        public Meeting(Meeting meeting, IEnumerable<Participant> participants) : this(meeting)
        {
            Participants = participants;
        }

        public Meeting(Meeting meeting, IEnumerable<Polls.Poll> pollResults) : this(meeting)
        {
            PollResults = pollResults;
        }
    }
}
